src/tools/rbd_mirror/LeaderWatcher.h

```diff
@@ -44,8 +88,45 @@ class LeaderWatcher : protected librbd::Watcher {
     }
   };
 
+  Mutex &m_lock;
+  Cond &m_cond;
   uint64_t m_notifier_id;
 
+  bool m_stopping = false;
```

Comment:
Can you merge these three into a single state enum? It seems like you shouldn't be doing all three at the same time.